core/trino-main/src/main/java/io/trino/cost/FilterStatsCalculator.java

```diff
@@ -423,9 +424,16 @@ private SymbolStatsEstimate getExpressionStats(Expression expression)
             return scalarStatsCalculator.calculate(expression, input, session, types);
         }
 
-        private OptionalDouble doubleValueFromLiteral(Type type, Literal literal)
+        private boolean isEffectivelyLiteral(Expression expression)
         {
-            Object literalValue = LiteralInterpreter.evaluate(plannerContext, session, getExpressionTypes(plannerContext, session, literal, types), literal);
+            return ExpressionUtils.isEffectivelyLiteral(plannerContext, session, expression);
+        }
+
+        private OptionalDouble doubleValueFromLiteral(Type type, Expression literal)
+        {
+            ExpressionInterpreter interpreter = new ExpressionInterpreter(literal, plannerContext, session, getExpressionTypes(plannerContext, session, literal, types));
+            Object literalValue = interpreter.optimize(NoOpSymbolResolver.INSTANCE);
+            checkArgument(!(literalValue instanceof Expression), "Expression is not literal: %s", literal);
```

Comment:
After validating expressions of the form `Cast(Literal)` in the `isEffectivelyLiteral()` method, and expressions being `Literal`s as in https://github.com/trinodb/trino/pull/10720, we can skip the validation here, and just use `interpreter.evaluate()`.

Reply:
switched to `evaluateConstantExpression`, it calls `ExpressionInterpreter`.evaluate` behind the scenes